test/cypress/e2e/multi-datasources/multi_datasources_enabled.spec.js

```diff
@@ -91,34 +90,83 @@ describe('Multi-datasources enabled', () => {
   });
 
-  it.skip('Checks Users Tab', () => {
+  it('Checks Users Tab', () => {
     cy.visit('http://localhost:5601/app/security-dashboards-plugin#/users');
     // Create an internal user in the remote cluster
     cy.contains('h3', 'Internal users');
     cy.contains('a', 'admin');
-    // TODO replace these with navigating to urls that get read to determine datasource, since these are flaky
+
+    closeToast();
+
+    // select remote data source
     cy.get('[data-test-subj="dataSourceSelectableContextMenuHeaderLink"]').click();
-    cy.contains('li.euiSelectableListItem', '9202').click();
+    cy.get('[title="9202"]').click();
+
+    // create a user on remote data source
     cy.get('[data-test-subj="create-user"]').click();
     cy.get('[data-test-subj="name-text"]').focus().type('9202-user');
     cy.get('[data-test-subj="password"]').focus().type('myStrongPassword123!');
     cy.get('[data-test-subj="re-enter-password"]').focus().type('myStrongPassword123!');
     cy.get('[data-test-subj="submit-save-user"]').click();
 
     // Internal user exists on the remote
-    cy.visit('http://localhost:5601/app/security-dashboards-plugin#/users');
+    cy.get('[data-test-subj="dataSourceSelectableContextMenuHeaderLink"]').should(
+      'contain',
+      '9202'
+    );
     cy.get('[data-test-subj="checkboxSelectRow-9202-user"]').should('exist');
 
     // Internal user doesn't exist on local cluster
     cy.get('[data-test-subj="dataSourceSelectableContextMenuHeaderLink"]').click();
-    cy.contains('li.euiSelectableListItem', 'Local cluster').click();
+    cy.get('[title="Local cluster"]').click();
+    cy.get('[data-test-subj="dataSourceSelectableContextMenuHeaderLink"]').should(
+      'contain',
+      'Local cluster'
+    );
     cy.get('[data-test-subj="checkboxSelectRow-9202-user"]').should('not.exist');
   });
+
+  it('Checks Permissions Tab', () => {
+    cy.visit('http://localhost:5601/app/security-dashboards-plugin#/permissions');
+    // Create a permission in the remote cluster
+    cy.contains('h3', 'Permissions');
+
+    closeToast();
+
+    // Select remote cluster
+    cy.get('[data-test-subj="dataSourceSelectableContextMenuHeaderLink"]').click();
+    cy.get('[title="9202"]').click();
+    cy.get('[data-test-subj="dataSourceSelectableContextMenuHeaderLink"]').should(
+      'contain',
+      '9202'
+    );
+
+    // Create an action group
+    cy.get('[id="Create action group"]').click();
+    cy.contains('div.euiFlexItem > button', 'Create from blank').click();
+    cy.get('[data-test-subj="name-text"]')
+      .focus()
+      .type('test_permission_ag', { force: true })
+      .should('have.value', 'test_permission_ag');
+    cy.get('[data-test-subj="comboBoxInput"]').focus().type('some_permission');
+    cy.get('[id="submit"]').click();
+
+    // Permission exists on the remote data source
+    cy.get('[data-text="Customization"]').click();
+    cy.contains('div.euiFilterSelect__items > button', 'Custom').click();
```

Comment:
Is there a way we can use data-test-subj instead? My concern with these types of selections is that if core adds something outside of our control it can affect this test.